.github/workflows/ci_tests.yml

```diff
@@ -0,0 +1,122 @@
+name: CI Tests
+
+on:
+  push:
+  pull_request:
+  schedule:
+    # run every Monday at 6am UTC
+    - cron: '0 6 * * 1'
+
+env:
+  SETUP_XVFB: True  # avoid issues if mpl tries to open a GUI window
+  TOXARGS: '-v'
+
+jobs:
+  ci-tests:
+    name: ${{ matrix.os }}, ${{ matrix.tox_env }}
+    runs-on: ${{ matrix.os }}
+    if: "!(contains(github.event.head_commit.message, '[skip ci]') || contains(github.event.head_commit.message, '[ci skip]'))"
+    strategy:
+      matrix:
+        name: [
+          'ubuntu-py36',
+          'ubuntu-py37',
+          'ubuntu-py38',
+          'macos-py38',
+          'windows-py38',
+          'ubuntu-py38-test',
+          'ubuntu-codestyle',
+          'ubuntu-build_docs',
+          'ubuntu-linkcheck',
+          'ubuntu-bandit',
+          'ubuntu-py36-test-alldeps-astropylts-numpy117',
+          'ubuntu-py37-test-alldeps-astropylts-numpy118',
+          'ubuntu-py38-test-alldeps-devdeps',
+        ]
```

Comment:
Is this actually needed? (the name: entry) - just curious!

Reply:
I think so.  The matrix is on the name entries, which are then specified in detail below in the `include` section.  This allows for more control to specify exactly what you want (and only what you want) in the matrix.  Please let me know if there is a better way!

Reply:
I found this to be the cleanest way to set up the matrix without having to duplicate the workflow steps.  I got the idea from the `pytest` repo (https://github.com/pytest-dev/pytest/blob/master/.github/workflows/main.yml)

Reply:
It would probably be an overoptimization, but I wonder whether it's possible to nest these, e.g. start with ubuntu-py36, then in a second run install all the dependencies and test again? Maybe it would only be useful for core astropy?

Reply:
Actions doesn't have stages like Travis, so you'd have to manually copy a new workflow.  The good news is that the workflow "fails fast" such that if any of the runs fail, the entire workflow fails and all jobs are stopped.

Reply:
@larrybradley - I didn't seem to need it here: 

https://github.com/matplotlib/pytest-mpl/blob/master/.github/workflows/python-package.yml#L15

which is why I was wondering why it was for photutils?

Reply:
🤔 I'm not sure I tried it without the names.  I'll do that.

Reply:
I remove the name section:  https://github.com/astropy/photutils/pull/1114.  Thanks, @astrofrog.